### PR TITLE
build: sveltekit (without routing changes)

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Use an opiniated configuration with SvelteKit and ViteJS. This brings best practice to the dapp, improve the DX and make the project easier to jump in for the community.

# Note

This PR introduces SvelteKit but keep current custom routing system.

# Changes

- replace `rollup` build with `SvelteKit` (which uses `vitejs` and ... `rollup`)
- update all JS import updated to ` import ... from $lib/...`
- ts config and compilation set to `strict: true`
- `deployment-configuration.json` is replaced by `.env` and `HACKING.md` documented accordingly
- update script that build `index.html` to "only" parse the CSP
- remove the custom script loader from `index.html` because Sveltekit injects the script loader
- move `index.html` content to `app.html`
- move previous `routes` to `lib/routes`
- create a single `+page.svelte` that render our custom routes
- use adapter static = prerender the application
- remove `BUILD` info from `dfx.json` - not used anymore
- remove `v2/index.html`

# Tests

- npm run dev / build :white_check_mark:
- npm run test / check / lint  :white_check_mark:
- CI :white_check_mark:
- deploy on a canister https://tqtu6-byaaa-aaaaa-aaana-cai.small13.dfinity.network/  :white_check_mark:

### CODE REVIEW

File: frontend/README.md [hunk 1]
```diff
@@ -1,57 +1,38 @@
-# NNS Dapp Frontend (Svelte)
+# create-svelte
```
Comment (lmuntaner) [inline frontend/README.md:1]: Should you revert this file?

File: frontend/package.json [hunk 2]
```diff
@@ -1,82 +1,74 @@
 {
-  "name": "@dfinity/nns-dapp",
-  "version": "1.0.0",
+  "name": "my-app",
```
Comment (lmuntaner) [inline frontend/package.json:2]: revert the name?

File: frontend/package.json [hunk 3]
```diff
@@ -1,82 +1,74 @@
...
+  "version": "0.0.1",
```
Comment (lmuntaner) [inline frontend/package.json:3]: Revert as well or even move to 2.0.0?

File: frontend/src/lib/components/accounts/AccountCard.svelte [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-  import AmountDisplay from "../ic/AmountDisplay.svelte";
-  import Identifier from "../ui/Identifier.svelte";
+  import AmountDisplay from "$lib/components/ic/AmountDisplay.svelte";
+  import Identifier from "$lib/components/ui/Identifier.svelte";
   import type { TokenAmount } from "@dfinity/nns";
   import AccountBadge from "./AccountBadge.svelte";
 
   export let account: Account;
-  export let showCopy: boolean = false;
+  export let showCopy = false;
```
Comment (lmuntaner) [inline frontend/src/lib/components/accounts/AccountCard.svelte:10]: why this change? I guess it's automatic, but this goes against the pattern we were using...

I like it like this, but I just want to make sure it's a conscious change.

File: frontend/static/assets/libs/marked.min.js [hunk 5]
```diff
@@ -4,5 +4,1873 @@
...
\ No newline at end of file
+function getDefaults() {
```
Comment (lmuntaner) [inline frontend/static/assets/libs/marked.min.js:7]: Maybe revert these changes?

File: frontend/tsconfig.spec.json [hunk 6]
```diff
@@ -1,6 +1,24 @@
...
+    "strict": false
+  },
+  "include": [
+    ".svelte-kit/ambient.d.ts",
+    "./vite.config.ts",
+    "./src/**/*.js",
+    "./src/**/*.ts",
+    "./src/**/*.svelte"
+  ]
+  // Path aliases are handled by https://kit.svelte.dev/docs/configuration#alias
```
Comment (lmuntaner) [inline frontend/tsconfig.spec.json:20]: Doesn't this break the JSON? I think to remember that comments are not well supported yet in JSON

**Review by lmuntaner — COMMENTED** [review]
Wow, not bad the changes! Thanks for the effort!

I added a few questions, but it looks very good.

File: frontend/README.md (hunk 1 above)
Comment (peterpeterparker) [reply frontend/README.md]: Well spotted. Done and (lightly) updated as well.

File: frontend/package.json (hunk 3 above)
Comment (peterpeterparker) [reply frontend/package.json]: Excellent point. Done. In `package-lock` as well.

File: frontend/src/lib/components/accounts/AccountCard.svelte (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/accounts/AccountCard.svelte:10]: SvelteKit opinionated eslint configuration.

I thought about overwritting it but, keeping the default makes the dapp easier to approach for the community therefore applied the change.

File: frontend/static/assets/libs/marked.min.js (hunk 5 above)
Comment (peterpeterparker) [reply frontend/static/assets/libs/marked.min.js]: Done. Also excluded "static/assets/libs" from prettier

File: frontend/tsconfig.spec.json (hunk 6 above)
Comment (peterpeterparker) [reply frontend/tsconfig.spec.json:20]: For some reason, not. That doesn't comes from me, out of the box SvelteKit. Removed the comments

**Review by lmuntaner — APPROVED** [review]
Let's get this in!

File: frontend/package.json (hunk 2 above)
Comment (peterpeterparker) [reply frontend/package.json]: done
